components/merkledb/src/views/tests.rs

```diff
@@ -847,3 +850,62 @@ fn multiple_patch() {
     let iter = index.iter();
     assert_eq!(index.len() as usize, iter.count());
 }
+
+#[test]
+fn valid_index_name() {
+    assert!(check_valid_name("index_name"));
+    assert!(check_valid_name("_index_name"));
+    assert!(check_valid_name("AinDex_name_"));
+    assert!(check_valid_name("core.index_name1Z"));
+    assert!(check_valid_name("configuration.indeX_1namE"));
+    assert!(check_valid_name("1index_Namez"));
```

Comment:
To make spell checker happy should be replaced for: 
```suggestion
    assert!(check_valid_name("1index_Name"));
```